src/mapproxyUtils/mapproxySeed.ts

```diff
@@ -242,17 +242,66 @@ export class MapproxySeed {
         this.logger.info('requested to skip uncached tiles');
         flags.push('--skip-uncached');
       }
+
       const cmd = $`mapproxy-seed ${flags}`;
+      // promise wrap to synchronized zx internal events with node run time event
+      await new Promise<void>((resolve, reject) => {
+        cmd.stdout
+          .on('data', (chunk: Buffer) => {
+            try {
+              this.seedProgressFunc(chunk);
+            } catch (error) {
+              // this section catching seeding errors and fail the task by kill child process of cmd
+              void cmd.kill('SIGHUP');
+              cmd.child.on('exit', () => {
+                reject(new Error((error as Error).message));
+              });
+            }
+          })
+          .on('end', () => {
+            this.logger.info(`Completed ${options.mode} task`);
+            resolve();
+          });
+      });
 
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-      for await (const chunk of cmd.stdout) {
-        // todo - implement on next phase progressPercentage calculate logic
-        const str: string = (chunk as Buffer).toString('utf8');
-        this.logger.debug(str);
-      }
+      // promise wrap zx event on case of internal crash - so the node will catch up and throw exception to main loop
+      const exitCode = await cmd.exitCode;
+      await new Promise<void>((resolve, reject) => {
+        cmd.catch((error) => {
+          if ((error as ProcessOutput).exitCode !== 0) {
```

Comment:
can't we log the error here (before the if)? (with jobId, ...)

Reply:
added